src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.SymbolComputer.cs

```diff
@@ -178,7 +178,9 @@ private static ImmutableArray<PortableExecutableReference> GetAllRelevantPeRefer
 
                 var originatingAssembly = _originatingSemanticModel.Compilation.Assembly;
                 var filter = CreateAggregatedFilter(cacheEntry);
-                var compilation = await project.GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
+                var compilation = project == _originatingDocument.Project
+                    ? _originatingSemanticModel.Compilation
+                    : await project.GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
doc plz.  this is also def a bit confusing.  do we doc anywhere that _originatingSemanticModel is potentially a speculative model?  if not, can we?

Reply:
➡️ Documented